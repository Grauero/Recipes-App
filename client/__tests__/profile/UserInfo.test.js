import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import toJSON from 'enzyme-to-json';

import UserInfo from '../../src/components/profile/UserInfo';

const props = {
  session: {
    getCurrentUser: {
      username: 'username',
      email: 'email',
      joinDate: 'Tue Feb 26 2019 17:30:35 GMT+0300 (Москва, стандартное время)',
      favorites: [{ _id: '_id', name: 'name' }]
    }
  }
};

let component;
let wrapper;

beforeEach(() => {
  wrapper = mount(
    <BrowserRouter>
      <UserInfo {...props} />
    </BrowserRouter>
  );

  component = wrapper.find(UserInfo);
});

afterEach(() => {
  wrapper.unmount();
});

it('matches snapshot', () => {
  expect(toJSON(component)).toMatchSnapshot();
});

it('formats Date correctly', () => {
  expect(
    component
      .find('p')
      .at(2)
      .html()
  ).toBe('<p>Join Date: 2/26/2019 at 5:30:35 PM</p>');
});

it('renders list of favorites if it exists', () => {
  expect(component.find('li').html()).toBe('<li><a href="/recipes/_id"><p>name</p></a></li>');
});

it('renders message if user did NOT have favorites', () => {
  const props = {
    session: {
      getCurrentUser: {
        username: 'username',
        email: 'email',
        joinDate: 'Tue Feb 26 2019 17:30:35 GMT+0300 (Москва, стандартное время)',
        favorites: []
      }
    }
  };
  wrapper = mount(
    <BrowserRouter>
      <UserInfo {...props} />
    </BrowserRouter>
  );

  expect(
    wrapper
      .find(UserInfo)
      .find('strong')
      .html()
  ).toBe('<strong>You have no favorites currently.</strong>');
});
