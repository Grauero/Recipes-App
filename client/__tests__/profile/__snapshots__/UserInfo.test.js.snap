// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<UserInfo
  session={
    Object {
      "getCurrentUser": Object {
        "email": "email",
        "favorites": Array [
          Object {
            "_id": "_id",
            "name": "name",
          },
        ],
        "joinDate": "Tue Feb 26 2019 17:30:35 GMT+0300 (Москва, стандартное время)",
        "username": "username",
      },
    }
  }
>
  <div>
    <h3>
      User Info
    </h3>
    <p>
      Username: 
      username
    </p>
    <p>
      Email: 
      email
    </p>
    <p>
      Join Date: 
      2/26/2019 at 5:30:35 PM
    </p>
    <ul>
      <h3>
        username
        's Favorites
      </h3>
      <li
        key="_id"
      >
        <Link
          replace={false}
          to="/recipes/_id"
        >
          <a
            href="/recipes/_id"
            onClick={[Function]}
          >
            <p>
              name
            </p>
          </a>
        </Link>
      </li>
    </ul>
  </div>
</UserInfo>
`;
